spring-kafka/src/test/kotlin/org/springframework/kafka/support/serializer/JacksonJsonDeserializerTests.kt

```diff
@@ -0,0 +1,29 @@
+package org.springframework.kafka.support.serializer
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertNull
+import tools.jackson.core.type.TypeReference
+import kotlin.text.Charsets.UTF_8
+
+class JacksonJsonDeserializerTests {
+    private val targetType = object : TypeReference<DeserializedData>() {}
+    private val deserializer = JacksonJsonDeserializer(targetType).ignoreTypeHeaders()
+
+    @Test
+    fun `Expect deserializing a non-null value to work`() {
+        val dataToDeserialize = """{"value":"test-data"}""".toByteArray(UTF_8)
+        val deserializedData = deserializer.deserialize("topic", dataToDeserialize)
+
+        assertEquals(DeserializedData("test-data"), deserializedData)
+    }
+
+    @Test
+    fun `Expect deserializing a null value to work`() {
+        val deserializedData = deserializer.deserialize("topic", null)
+        assertNull(deserializedData)
+    }
+
+    private data class DeserializedData(val value: String)
+
+}
```

Comment:
New line in the end of file, please.
`@since 4.0.1`.
Your name to the `@author` list of all the affected classes.
Copy a Copyright from other similar Kotlin classes.